src/main/javascript/crypto/src/libs/Eip712Validator.ts

```diff
@@ -0,0 +1,118 @@
+import {AttestedObject} from "./AttestedObject";
+import {UseToken} from "../asn1/shemas/UseToken";
+import {XMLconfigData} from "../data/tokenData";
+import {KeyPair} from "./KeyPair";
+import {Ticket} from "../Ticket";
+import {Eip712DomainInterface, SignatureUtility} from "./SignatureUtility";
+import {hexStringToUint8} from "./utils";
+const url = require('url');
+
+export class Eip712Validator {
+    private XMLConfig: any;
+    protected domain: string;
+    // protected acceptableTimeLimitMs: number;
+    // public DEFAULT_TIME_LIMIT_MS = 1000 * 60 * 20; // 20 min
+
+    constructor() {
+        this.XMLConfig = XMLconfigData;
+    }
+
+    static stringIsAValidUrl(domain: string): boolean {
+        let parsedUrl;
+
+        try {
+            parsedUrl = new URL(domain);
+        } catch (e) {
+            console.log('cant construct url. Error:' + e);
+            return false;
+        }
+
+        return parsedUrl.protocol === "http:" || parsedUrl.protocol === "https:";
+    };
+
+    setDomainAndTimout(domain: string){
+        if (!Eip712Validator.stringIsAValidUrl(domain)) throw new Error('wrong domain');
+        this.domain = domain;
+    }
+
+    setDomain(domain: string){
+        if (!Eip712Validator.stringIsAValidUrl(domain)) throw new Error('wrong domain');
+        this.domain = domain;
+    }
+
+    getDomain(): string{
+        return this.domain;
+    }
+
+    validateRequest(jsonInput: string) {
+        try {
+            let authenticationData = JSON.parse(jsonInput);
+
+            let authenticationRootNode = JSON.parse(authenticationData.jsonSigned);
+
+            // console.log(authenticationRootNode);
+
+            let eip712Domain = authenticationRootNode.domain;
+            let eip712Message = authenticationRootNode.message;
+
+            console.log('eip712Domain');
+            console.log(eip712Domain);
+            console.log(eip712Message);
+
+            let attestedObject = this.retrieveAttestedObject(eip712Message);
+
+            // TODO implement
+            return this.validateDomain(eip712Domain)
```

Comment:
Does this need to be implemented here or is it _always_ implemented/used in the classes using this class?

Reply:
are you asking for "validateDomain()" method? its defined next to that request.